### PR TITLE
Add checkov, terraform-fmt, terraform-validate reusable workflows

### PR DESCRIPTION
**Pull request by kuzmacska**

### CODE REVIEW

File: .github/workflows/terraform-checkov-reusable.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    inputs:
+      soft_fail:
+        description: 'Do not return an error code if there are failed checks'
+        required: false
+        type: boolean
+        default: true
+
+jobs:
+  checkov-job:
+    runs-on: self-hosted
```
Comment (akaRem) [inline .github/workflows/terraform-checkov-reusable.yml:14]: Should we parameterise this?

File: .github/workflows/terraform-checkov-reusable.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    name: checkov-action
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Run Checkov action
+        id: checkov
+        uses: bridgecrewio/checkov-action@bb0220b84d1cf29490e505203566ed34451c816d
```
Comment (akaRem) [inline .github/workflows/terraform-checkov-reusable.yml:21]: They have tags. Why commit sha?

File: .github/workflows/terraform-checkov-reusable.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          directory: .
+          quiet: true # optional: display only failed checks
+          soft_fail: ${{ inputs.soft_fail }} # optional: do not return an error code if there are failed checks
+          framework: terraform # optional: run only on a specific infrastructure {cloudformation,terraform,kubernetes,all}
+          output_format: cli # optional: the output format, one of: cli, json, junitxml, github_failed_only
+          download_external_modules: false # optional: download external terraform modules from public git repositories and terraform registry
+#          log_level: DEBUG # optional: set log level. Default WARNING
```
Comment (akaRem) [inline .github/workflows/terraform-checkov-reusable.yml:29]: I think, it makes sense to parameterise, will make life of consumers easier

File: .github/workflows/terraform-fmt-reusable.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+name: Terraform fmt
+
+on:
+  workflow_call:
+    inputs:
+      poweruser_name:
```
Comment (akaRem) [inline .github/workflows/terraform-fmt-reusable.yml:6]: Is it a convention? in git it's called commiter_name, commiter_email

File: .github/workflows/terraform-fmt-reusable.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        run: |
+          sudo chown -R $USER:$USER .
+          git config --global user.name '${{ inputs.poweruser_name }}'.
+          git config --global user.email '${{ inputs.poweruser_email }}'
+          git add -- ./\*.tf
+          if ! git diff-index --quiet --cached HEAD; then
+            git commit -m "[AUTOMATED] Terraform fmt: format the codebase recursively"
+          fi
+      - name: Push changes for terraform fmt
+        uses: ad-m/github-push-action@552c074ed701137ebd2bf098e70c394ca293e87f
```
Comment (akaRem) [inline .github/workflows/terraform-fmt-reusable.yml:48]: same here

File: .github/workflows/terraform-validate-reusable.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+          git config --global url."https://foo:${{ secrets.GIT_CLONE_TOKEN }}@github.com".insteadOf "ssh://git@github.com"
+      - name: Terraform Validate modules
+        run: |
+          cd ${{ inputs.modules_subpath }}
+          if (ls *.tf) 
+          then
+            echo "This is not a modules directory. Please use validation at the project level after initialization or choose a modules directory"
+            exit 1
+          else
+            for dir in $(ls -d */); do (cd "$dir" && pwd && terraform init -backend=false && terraform validate); done
```
Comment (akaRem) [inline .github/workflows/terraform-validate-reusable.yml:38]: https://www.shellcheck.net/wiki/SC2045

but find/exec would be even beter

File: .github/workflows/terraform-validate-reusable.yml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      - name: Configure ability to fetch terraform modules from private repos
+        run: |
+          git config --global url."https://foo:${{ secrets.GIT_CLONE_TOKEN }}@github.com".insteadOf "ssh://git@github.com"
+      - name: Terraform Validate modules
+        run: |
+          cd ${{ inputs.modules_subpath }}
+          if (ls *.tf)
```
Comment (akaRem) [inline .github/workflows/terraform-validate-reusable.yml:33]: https://www.shellcheck.net/wiki/SC2035

**Review by akaRem — CHANGES_REQUESTED** [review]

File: .github/workflows/terraform-fmt-reusable.yml (hunk 4 above)
Comment (kuzmacska) [reply .github/workflows/terraform-fmt-reusable.yml:6]: this is human friendly name

File: .github/workflows/terraform-fmt-reusable.yml (hunk 5 above)
Comment (kuzmacska) [reply .github/workflows/terraform-fmt-reusable.yml:48]: last tag in 2020

File: .github/workflows/terraform-checkov-reusable.yml (hunk 1 above)
Comment (kuzmacska) [reply .github/workflows/terraform-checkov-reusable.yml:14]: no, this is already generic enough

**Review by akaRem — APPROVED** [review]

**Pull request merged**